src/test/java/de/tum/in/www1/artemis/TestRepositoryResourceIntegrationTest.java

```diff
@@ -0,0 +1,200 @@
+import org.springframework.util.LinkedMultiValueMap;
+
+import de.tum.in.www1.artemis.domain.Course;
+import de.tum.in.www1.artemis.domain.FileType;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.repository.ProgrammingExerciseRepository;
+import de.tum.in.www1.artemis.util.*;
+import de.tum.in.www1.artemis.web.rest.dto.FileMove;
+import de.tum.in.www1.artemis.web.rest.dto.RepositoryStatusDTO;
+
+public class TestRepositoryResourceIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    private final String testRepoBaseUrl = "/api/test-repository/";
+
+    @Autowired
+    private DatabaseUtilService database;
+
+    @Autowired
+    private RequestUtilService request;
+
+    @Autowired
+    ProgrammingExerciseRepository programmingExerciseRepository;
+
+    private Course course;
+
+    private ProgrammingExercise exercise;
+
+    private String currentLocalFileName = "currentFileName";
+
+    private String currentLocalFileContent = "testContent";
+
+    private String currentLocalFolderName = "currentFolderName";
+
+    private String newLocalFileName = "newFileName";
+
+    private String newLocalFolderName = "newFolderName";
+
+    LocalRepository testRepo = new LocalRepository();
+
+    @BeforeEach
+    public void setup() throws Exception {
+        database.addUsers(0, 0, 1);
+        course = database.addEmptyCourse();
+        exercise = ModelFactory.generateProgrammingExercise(ZonedDateTime.now().minusDays(1), ZonedDateTime.now().plusDays(7), course);
+        testRepo.configureRepos("testLocalRepo", "testOriginRepo");
+
+        // add file to the repository folder
+        Path filePath = Paths.get(testRepo.localRepoFile + "/" + currentLocalFileName);
+        var file = Files.createFile(filePath).toFile();
+        // write content to the created file
+        FileUtils.write(file, currentLocalFileContent);
+
+        // add folder to the repository folder
+        filePath = Paths.get(testRepo.localRepoFile + "/" + currentLocalFolderName);
+        var folder = Files.createDirectory(filePath).toFile();
+
+        var testRepoUrl = new GitUtilService.MockFileRepositoryUrl(testRepo.originRepoFile);
+        exercise.setTestRepositoryUrl(testRepoUrl.toString());
+        doReturn(gitService.getRepositoryByLocalPath(testRepo.localRepoFile.toPath())).when(gitService).getOrCheckoutRepository(testRepoUrl.getURL(), true);
+    }
+
+    @AfterEach
+    public void tearDown() throws IOException {
+        database.resetDatabase();
+        reset(gitService);
+        LocalRepository.resetLocalRepo(testRepo);
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void shouldGetFiles() throws Exception {
+        programmingExerciseRepository.save(exercise);
+        var files = request.getMap(testRepoBaseUrl + exercise.getId() + "/files", HttpStatus.OK, String.class, FileType.class);
```

Comment:
check that the files actually exists, some of them might get filtered by the way?

Reply:
i added a check, which checks if the files actually exist